frontend/src/Components/roadDetails/TopBar.tsx

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import { Link, useNavigate } from 'react-router-dom';
+
+interface TopBarProps {
+  setShowMiddleArea2: React.Dispatch<React.SetStateAction<boolean>>;
```

Comment:
Add comment to describe the variable and use a non-specific name, for instance:
```suggestion
  /** A callback that is called when the toogle button change value */
  isToggleOn: React.Dispatch<React.SetStateAction<boolean>>;
```